src/qililab/calibration/calibration_node.py

```diff
@@ -0,0 +1,668 @@
+        if len(outputs_lines) != 1:
+            logger.error("No output or various outputs found in notebook %s.", self.nb_path)
+            raise IncorrectCalibrationOutput(f"No output or various outputs found in notebook {self.nb_path}.")
+
+        # When only one line of outputs, use that one:
+        return self._from_logger_string_to_output_dict(outputs_lines[0], self.nb_path)
+
+    def _from_logger_string_to_output_dict(self, logger_string: str, input_path: str) -> dict:
+        """Returns the output dictionary from a logger output string. Raises errors if the ouput doesn't follow the expected format.
+
+        Args:
+            logger_string (str): The logger string containing the output dictionary to extract.
+            input_path (str): Path of the notebook that generated the output, to raise errors.
+
+        Returns:
+            dict: The output dictionary of the file execution.
+
+        Raises:
+            IncorrectCalibrationOutput: In case no outputs, incorrect outputs or multiple outputs where found. Incorrect outputs are those that do not contain `check_parameters` or is empty.
+        """
+        logger_splitted = logger_string.split(logger_output_start)
+
+        # In case something unexpected happened with the output we raise an error
+        if len(logger_splitted) != 2:
+            logger.error("No output or various outputs found in notebook %s.", input_path)
+            raise IncorrectCalibrationOutput(f"No output or various outputs found in notebook {input_path}.")
+
+        # This next line is for taking into account other encodings, where special characters get `\\` in front.
+        clean_data = logger_splitted[1].split("\\n")[0].replace('\\"', '"')
+
+        logger_outputs_string = clean_data.split("\n")[0]
+        out_dict = json.loads(logger_outputs_string)
+
+        if "check_parameters" not in out_dict or out_dict["check_parameters"] == {}:
+            logger.error(
+                "Aborting execution. No 'check_parameters' dictionary or its empty in the output cell implemented in %s",
+                input_path,
+            )
+            raise IncorrectCalibrationOutput(
+                f"Empty output found in {input_path}, output must have key and value 'check_parameters'."
+            )
+        return out_dict
+
+    def _add_string_to_checked_nb_name(self, string_to_add: str, timestamp: float) -> None:
+        """Adds a string to the checked notebook name.
+
+        Args:
+            string_to_add (str): The string to append to the notebook name.
+            timestamp (float): The timestamp to use for the new notebook execution name.
+        """
+        path = os.path.join(self.nb_folder, self.node_id)
+        timestamp_path = self._create_notebook_datetime_path(path, timestamp).split(".ipynb")[0]
+
+        os.rename(f"{timestamp_path}.ipynb", f"{timestamp_path}_{string_to_add}.ipynb")
+
+
+def export_calibration_outputs(outputs: dict) -> None:
```

Comment:
Can we shorten a bit the name of this function? Maybe to `export_outputs`?

Reply:
Sure yes! What do you think about`export_nb_outputs`? 👀